app/controllers/application_controller.rb

```diff
@@ -1,20 +1,46 @@
-# typed: true
+# typed: false
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  extend T::Sig
+
   include DeviseRemoteUser::ControllerBehavior
 
   # Smartly redirect user back to URL they requested before authenticating
   # From: https://github.com/heartcombo/devise/wiki/How-To:-Redirect-back-to-current-page-after-sign-in,-sign-out,-sign-up,-update
   before_action :store_user_location!, if: :storable_location?
+  # before_action :authenticate_user!
+
+  rescue_from ActionPolicy::Unauthorized, with: :deny_access
+
+  sig { returns(T.nilable(User)) }
+  def current_user
+    super.tap do |cur_user|
+      break unless cur_user
+
+      cur_user.groups = ldap_groups
+    end
+  end
 
   private
 
+  sig { returns(T::Array[String]) }
+  def ldap_groups
+    return request.env['eduPersonEntitlement'].split(';') if request.env['eduPersonEntitlement']
+
+    []
```

Comment:
d'oh, just realized this line's code path isn't tested, happy to revisit if people want.

Reply:
also, fwiw, i suspect the lack of diff coverage on this line is what's failing the build

Reply:
fixed